src/replacements.js

```diff
@@ -0,0 +1,188 @@
+const Gio = imports.gi.Gio;
+const EosGates = imports.eos_gates;
+
+function generateEndlessInstallerEntry() {
+    let entry = {
+        regex: {
+            windows: /endless-installer.*.exe/
+        },
+        overrideHelpMessage: _("You are already running Endless OS")
+    };
+
+    // If we detect the presence of eos-installer.desktop, inform
+    // the user on how to actually install Endless OS.
+    let desktopInfo = Gio.DesktopAppInfo.new('eos-installer.desktop');
+    if (desktopInfo) {
+        entry.overrideHelpMessage = _("You are already running Endless OS from live media");
+        entry.desktopInfo = desktopInfo,
+        entry.replacementInfo = {
+            appName: _("Endless OS Installer"),
+            description: _("You can install Endless OS by running the Endless OS Installer.")
+        }
+    }
+
+    return entry;
+}
+
+function definitions() {
+    return [
+        {
+            regex: {
+                windows: /spotifysetup.*.exe/i
+            },
+            appName: _("Spotify"),
+            flatpakInfo: { remote: 'eos-apps', id: 'com.spotify.Client' }
```

Comment:
Unfortunately, we have to pass the fully resolved ID to gnome-software. That said, it might be possible to look it up with Flatpak beforehand so that we don't have to specify it here.